Update docs v1.4 (#279)
AY2021S1-CS2103T-T10-4/tp: docs/DeveloperGuide.md

### Managing Students

#### Adding a student

> 1. Adding a student.

Reviewer: ```suggestion
1. Adding a student
```

   1. Prerequisites: None.

   1. Test case: `add-student n\Nancy Drew t\nancydrew e\nancy@drew.com tag\A star tag\student` \
      Expected: The student with the fields entered should be added into the displayed student list. The tags `A star` and `student` should be rendered.

   1. Test case: `add-student n\Eddie Davis t\davis_ed e\eddie@basie.com` \
     Expected: The student with the fields entered should be added into the displayed student list. There should be no tags rendered.

   1. Other incorrect commands to try:
      * `add-student`
      * `add-student n\Hendrix`
      Expected: No student is added. Error details shown in the status message.

#### Editing a student

> 1. Editing a student.

Reviewer: ```suggestion
1. Editing a student
```

   1. Prerequisites: There are at least 2 student currently displayed.

   1. Test case: `edit-student 1 n\Jim Hetfield` \
      Expected: The name of the first student is changed to `Jim Hetfield`.

   1. Test case: `edit-student 2 tag\` \
      Expected: All tags of the second student are deleted.

   1. Other incorrect commands to try:
      * `edit-student`
      * `edit-student 0 n\Fiona Orange`
      Expected: No student is edited. Error details shown in the status message.

#### Deleting a student

> 1. Deleting a student while all students are being shown.

Reviewer: ```suggestion
1. Deleting a student while all students are being shown
```

      size of student list)<br>
      Expected: No student is deleted. Error details shown in the status message.

#### Finding a student

> 1. Finding a student.

Reviewer: ```suggestion
1. Finding a student
```

   1. Prerequisites: Student names that contain `Alex` and `Yu`. The sample data of Tutor's Pet contains such names and can be used.

   1. Test case: `find-student alex` \
      Expected: All students whose name contains `alex` (case-insensitive) should be displayed.

   1. Test case: `find-student alex yu` \
      Expected: All students whose name contains `alex` or `yu` (case-insensitive) should be displayed.

>    1. Incorrect find command to try: `find-student`. \
>       Expected: The displayed student list does not get updated. Error details shown in the status message.

Reviewer 2: Perhaps it would be better to keep the format same as the previous few?
```suggestion  
    1. Other incorrect commands to try:
       * `find-student`
       Expected: The displayed student list does not get updated. Error details shown in the status message.
```

### Managing Lessons

#### Editing a lesson

> 1. Editing a lesson while all lessons are displayed

Reviewer 2: ```suggestion
1. Editing a lesson while all lessons are displayed.
```

Reviewer: I think most of existing instructions do not include a fullstop here?
